sysdeps/nacl/ioctl.c

```diff
@@ -0,0 +1,37 @@
+#include <errno.h>
+#include <stdarg.h>
+#include <sys/ioctl.h>
+
+#include <irt_syscalls.h>
+
+int __ioctl (int fd, unsigned long request, ...) {
+
+    if (fd < 0) {
+      __set_errno (EBADF);
+      return -1;
+    }
+
+    int arg = 0;
+    void* arg_ptr = &arg;
+    int result;
+
+    if (request == FIONBIO) {
```

Comment:
is FIONBIO included from the ioctl header?

Reply:
Yes. It is included from` \sysdeps\unix\bsd\bsd4.4\bits\ioctls.h` which is inclued through `<sys/ioctl.h>`

Reply:
hmm I believe its actually coming from the x86 header which pulls in some asm definitions. Either way its there. Cool!